src/computation/1_computation_parallel.cpp

```diff
@@ -0,0 +1,379 @@
+    }
+    
+    pressureSolver_ = std::make_unique<RedBlack>(discretization_, settings_.epsilon, settings_.maximumNumberOfIterations); 
+
+    // misc
+    dt_ = 0.;
+
+    // initialize output writer
+    outputWriterParaviewParallel_ = std::make_unique<OutputWriterParaviewParallel>(discretization_);
+    outputWriterTextParallel_     = std::make_unique<OutputWriterTextParallel>(discretization_);
+}
+
+void ComputationParallel::runSimulation()
+{
+    double currentTime = 0;
+    int sec = 0;
+
+    std::cout << "+++++++++++++++++++++++" << std::endl;
+    std::cout << "Starting at time: " << currentTime << std::endl;
+    std::cout << "+++++++++++++++++++++++" << std::endl;
+
+    // the steps correspond to the steps in our algorithm in the overleaf or docs/numsim-algos.tex
+    while (currentTime < settings_.endTime)
+    {
+        // step 1: set the boundary values
+        applyBoundaryValues();
+        std::cout << "Applied boundary values for u/v and F/G." << std::endl;
+
+        // step 2: compute time step width
+        computeTimeStepWidth();
+
+        std::cout << "Computed time step width: " << dt_ << std::endl;
+
+        // set the dt such that the simulation stops exactly on endTime
+        if (currentTime + dt_ > settings_.endTime)
+        {
+            dt_ = settings_.endTime - currentTime;
+
+            std::cout << std::endl;
+            std::cout << "Final time step!" << std::endl;
+        } 
+        // set the dt such that the simulation steps exactly over full multiples of 1s (for the output)
+        if (std:floor(currentTime + dt) == sec +1.)
+        {
+            dt_ = (double) sec - currentTime;
+        }
+        currentTime += dt_;
+
+        std::cout << "+++++++++++++++++++++++" << std::endl;
+        std::cout << "current Time: " << currentTime << std::endl;
+        std::cout << "+++++++++++++++++++++++" << std::endl;
+
+        // step 4: calculate F, G with first setting the boundary conditions of F, G (step 3)
+        computePreliminaryVelocities();
+
+        std::cout << "Computed preliminary velocities" << std::endl;
+
+        // step 5: compute the right hand side of the pressure equation
+        computeRightHandSide();
+
+        std::cout << "Computed right hand side" << std::endl;
+
+        // step 6: solve the pressure equation
+        computePressure();
+
+        std::cout << "Computed presure" << " from process number " << partitioning_.ownRankNo() << std::endl;
+
+        // step 7: calculate the final velocities
+        computeVelocities();
+
+        std::cout << "Computed velocities" << std::endl;
+
+        // step 9: write output
+        if (std:floor(currentTime + dt) == sec +1.)
+        {
+            outputWriterParaview_->writeFile(currentTime);
+            outputWriterText_->writeFile(currentTime);
+            sec = sec +1;
+        }
+
+    }
+}
+
+void ComputationParallel::computeTimeStepWidth()
+{
+    double boundary_diffusion = 0.;
```

Comment:
can we call the `computeTimeStepWidth()` Method of the parent class instead of this copy paste to reduce the code length?

Reply:
Yes, you might change that